File: aries-backchannels/dotnet/server/Controllers/AgentStatusController.cs
using Microsoft.AspNetCore.Mvc;

namespace DotNet.Backchannel.Controllers
{
    [Route("agent/command/status")]
    [ApiController]
    public class AgentStatusController : ControllerBase
    {
        [HttpGet]
        public IActionResult AgentStatus()
        {
            // NOTE: Because the agent runs inside the backchannel (not separate processes)
            // The agent is active as long as the backchannel is active
            var active = true;

            if (active) return Ok(new { status = "active" });
            else return StatusCode(418, new { status = "inactive" });
        }
    }
}

File: aries-backchannels/javascript/server/src/controllers/AgentStatusController.ts
import { Controller, Get, Res } from "@tsed/common";
import packageJson from '@aries-framework/core/package.json'

@Controller("/agent/command")
export class AgentStatusController {
  @Get("/status")
  getStatus(@Res() response: Res) {
    // NOTE: Because the agent runs inside the backchannel (not separate processes)
    // The agent is active as long as the backchannel is active
    const active = true;

    if (!active) {
      response.status(418);
      return {
        status: "inactive",
      };
    }

    return {
      status: "active",
    };
  }

  @Get("/version")
  getVersion(@Res() response: Res) {
    const [version] = packageJson.version.split("+");
    return version
  }
}
